src/Projection/InMemoryProjectionManager.php

```diff
@@ -192,4 +165,26 @@ public function fetchProjectionState(string $name): array
 
         return $this->projections[$name]->getState();
     }
+
+    private function fetchProjectionNamesRegex(?string $filter, int $limit): array
+    {
+        if (null === $filter) {
+            throw new Exception\InvalidArgumentException('No regex pattern given');
+        }
+
+        set_error_handler(function($errorNo, $errorMsg): void {
+            throw new Exception\RuntimeException($errorMsg);
+        });
+
+        try {
+            $result = preg_grep("/$filter/", array_keys($this->projections));
+
+            return array_slice($result, 0, $limit);
```

Comment:
Your previous code was also unsorted. So I am not sure if it was expected to sort.